### DISCUSSION TITLE
Active Maintenance

### DISCUSSION THREAD

**Discussion by bosd** [joshreve/dactyl-keyboard#119] [General]
Hi @joshreve Are you actively maintaining this fork??
It would be a pity if not, because this is one of the better known forks.

Yet the reality is the last merge is dated 8 september 2021.

**Comment by joshreve** [comment]
I wish I had time to.  I tried to keep up for a while, but I honestly got lucky that I had a period of enough free time to make it in the first place.   This was a passion project when I had a more relaxed period for work during COVID.  I believe @bullwinkle3000 has the best active fork of this.  We have some differences in approach, but he keeps his solid.  He is an active seller so he keeps improving it.  We also chat about the repo from time to time.

I'm hoping to get back to it someday, but I realize I've been saying that since Sept 2021 apparently.  I still love seeing what people make from this.  If anyone were interested in working on this fork directly with me, I'm definitley open to it.  I have some time to work on it, just not nearly enough to keep up and really maintain it.

So, my unfortunate answer is, I am not actively maintaining it, but hope to again someday either when someone shows interest in working on it directly or when work and family life slow down a bit.

**Reply by joshreve** [reply]
Thank you for asking this.  I really enjoy this repo and I'm going to try to set aside an hour a week to work on it.  Not much, but perhaps enough to stabilize some of the issues and revitalize it.  I also realize I need to be more merciless with pull request acceptance.  I tried to accommodate too many different options and it made upkeep even more challenging.  I need to keep it focused to the core capabilities and let others add extra interfaces or non-standard tweaks on forks.

**Comment by bosd** [comment]
> I tried to accommodate too many different options and it made upkeep even more challenging. I need to keep it focused to the core capabilities and let others add extra interfaces or non-standard tweaks on forks.

Personally I actually kinda like the "master" repo which accommodates a lot of the functions. 
Merging literally ALL would be too much indeed, as there are some very exotic ones.

I get it, that maintenance can be quite hard, as there is no now form of unittests / checks being made on the PR's.

**Reply by joshreve** [reply]
I'm still planning to keep as many geometry choices as possible, it is all of the workflows and setups (auto installs, bash scripts, windows batch files, docker, etc) that are more problematic to maintain.  Part of it is cadquery, which has been problematic from the start.  I may actually just switch to a pyOCCT helper.  

I am contemplating how to do sparse unittests as well.  The geometry aspect makes it a bit less straightforward, but even a check that I can geometrically recreate the exact original dactyl would go a long way, perhaps using a subtract and volume check or something else.  I'll have to think more on that.

Anyway, I've started refamiliarizing myself with it and hope to start some cleanup soon.  I also have to decide if I move the current main to a "legacy" branch, use release tags, or something similar since I will likley want to make some breaking changes to get it back on track.  The alternative would be a fresh repo.

**Reply by bosd** [reply]
@joshreve Any update on this?

**Reply by joshreve** [reply]
@bosd, Unfortunately my free time for these endevours has been extremely limited in the past few years.  Kids, promotions, aging parents, etc.  Mostly good stuff, but not much hobby time left.

I would love to get back to it as I think this has been by far my favorite hobby project, but I'm sorry to say, I don't know when that will be.
